idl/github.com/uber/cadence/shared.thrift

```diff
@@ -500,6 +519,30 @@ struct ExternalWorkflowExecutionCancelRequestedEventAttributes {
   30: optional WorkflowExecution workflowExecution
 }
 
+struct SignalExternalWorkflowExecutionInitiatedEventAttributes {
+  10: optional i64 (js.type = "Long") decisionTaskCompletedEventId
+  20: optional string domain
+  30: optional WorkflowExecution workflowExecution
+  40: optional string signalName
+  50: optional binary input
+  60: optional binary control
+}
+
+struct SignalExternalWorkflowExecutionFailedEventAttributes {
+  10: optional SignalExternalWorkflowExecutionFailedCause cause,
+  20: optional i64 (js.type = "Long") decisionTaskCompletedEventId
+  30: optional string domain
+  40: optional WorkflowExecution workflowExecution
+  50: optional i64 (js.type = "Long") initiatedEventId
+  60: optional binary control
+}
+
+struct ExternalWorkflowExecutionSignalRequestedEventAttributes {
```

Comment:
I will probably not change this, because the event type is `EventTypeExternalWorkflowExecutionSignalRequested`. If I rename this to ExternalWorkflowExecutionSignaledEventAttributes, then I probably need to change the event name to `EventTypeExternalWorkflowExecutionSignaled`, but we try to be consistent with SWF so wouldn't change event name.

I'm still open minded, let me know if you still prefer new name

Reply:
SWF is using ExternalWorkflowExecutionSignaled.